models/deepseek/v4-pro/expert_shared.py

```
@@ -309,25 +309,32 @@ def golden_expert_shared(tensors):
+    # fp32 rounding error that flips bf16 last-bit ties on the final cast; mirror
+    # the kernel's "accumulate-int32, then one fp32 dequant mul" order to match.
     x_local_i8 = tensors["x_local_i8"]                       # [T, D] int8
     x_local_scale_dq = tensors["x_local_scale_dq"].float()   # [T, 1]
-    x_local = x_local_i8.float() * x_local_scale_dq
-    sw1 = dequant_w(tensors["shared_w1"], tensors["shared_w1_scale"].float())
-    sw3 = dequant_w(tensors["shared_w3"], tensors["shared_w3_scale"].float())
-    sw2 = dequant_w(tensors["shared_w2"], tensors["shared_w2_scale"].float())
-
-    sh_gate = x_local @ sw1.T
-    sh_up = x_local @ sw3.T
+    w1_i8 = tensors["shared_w1"]; w1_scale = tensors["shared_w1_scale"].float()   # [MOE_INTER, D], [MOE_INTER]
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)